overlay: Export metadata for container and image in docker-inspect

rhvgoyal opened the pull request:
Fixes #14099 

Export metadata for container and image in docker-inspect when overlay
graphdriver is in use. Right now it is done only for devicemapper graph
driver.

Signed-off-by: Vivek Goyal vgoyal@redhat.com

LK4D4 commented on daemon/graphdriver/overlay/overlay.go:
```diff
@@ -168,7 +168,35 @@ func (d *Driver) Status() [][2]string {
+	upperDir := path.Join(dir, "upper")
+	workDir := path.Join(dir, "work")
+	mergedDir := path.Join(dir, "merged")
+	metadata["lowerDir"] = lowerDir
```
any particular reason for using camelcase with lower first? in devicemapper simple CamelCase is used.

rhvgoyal replied:
@LK4D4 

That's a good point. I did not pay attention to it. As other fields in docker inspect output seem to be CamleCase, I will change these as well.

rhvgoyal commented:
@LK4D4 refreshed the patch with CamelCase. PTAL.

LK4D4 commented on daemon/graphdriver/overlay/overlay.go:
```diff
@@ -168,7 +168,35 @@ func (d *Driver) Status() [][2]string {
+		return nil, err
+	}
+
+	lowerDir := path.Join(d.dir(string(lowerId)), "root")
```
I think you can assign directly to map without intermediate variables.

rhvgoyal committed:
overlay: Export metadata for container and image in docker-inspect

Export metadata for container and image in docker-inspect when overlay
graphdriver is in use. Right now it is done only for devicemapper graph
driver.

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>

rhvgoyal commented:
@LK4D4 Got rid of local variables and directly assigned to map.

LK4D4 commented:
@rhatdan Thanks! LGTM

rhvgoyal commented:
@LK4D4 I am @rhvgoyal and not @rhatdan. :-)

LK4D4 commented:
@rhvgoyal haha :) Those rh* guys...

rhatdan commented:
I like to take credit for @rhvgoyal work though.  :^)

jfrazelle commented:
Cool thanks LGTM
